### PR TITLE
fix: initializes a ghost user by default to replace that the users that has been deleted

### LINKED ISSUE #3317: Ghost 用户在文章列表可被点击，并提示“资源不存在”
### 是什么版本出现了此问题？

release-2.2

### 使用的什么数据库？

PostgreSQL

### 使用的哪种方式部署？

Docker Compose

### 在线站点地址

_No response_

### 发生了什么？

操作步骤：
1. 新建一个用户
2. 使用这个用户发布一篇文章
3. 删除这个用户，文章发布人变为 ghost

image

4. 在文章列表点击 ghost 用户，将会跳转到用户页面，会提示“资源不存在”

image

/kind bug
/area console

### 相关日志输出

_No response_

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by JustinLiang522**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

系统默认初始化一个 ghost 用户，用于表示用户已删除。

#### Which issue(s) this PR fixes:

Fixes #3317

```release-note
NONE
```

### CODE REVIEW

File: src/main/java/run/halo/app/content/impl/PostServiceImpl.java [hunk 1]
```diff
@@ -224,15 +224,15 @@ private Flux<Contributor> listContributors(List<String> usernames) {
             return Flux.empty();
         }
         return Flux.fromIterable(usernames)
-            .flatMap(username -> client.fetch(User.class, username))
+            .flatMap(username -> client.fetch(User.class, username)
+                .switchIfEmpty(client.fetch(User.class, "ghost")))
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/content/impl/PostServiceImpl.java:228]: 建议用 `Mono.defer(() -> client.fetch(User.class, "ghost"))` 包裹。这样之后不会立即调用 client.fetch 方法。

Comment (JustinLiang522) [reply src/main/java/run/halo/app/content/impl/PostServiceImpl.java:228]: 像这样吗？
```
.flatMap(username -> {
                var ghostDefferedMono = Mono.defer(() -> client.fetch(User.class, "ghost"));
                return client.fetch(User.class, username)
                    .switchIfEmpty(ghostDefferedMono);
            })
```

Comment (JohnNiang) [reply src/main/java/run/halo/app/content/impl/PostServiceImpl.java:228]: 更准确点儿可以这样：

```java
.switchIfEmpty(Mono.defer(() -> client.fetch(User.class, "ghost")))
```

**Review by guqing — COMMENTED** [review]
/lgtm

**Review by JohnNiang — APPROVED** [review]
/approve

**Pull request merged**